index_settings_test.go

```diff
@@ -1332,14 +1332,59 @@ func TestIndex_UpdateFilterableAttributes(t *testing.T) {
 			args: args{
 				UID:    "indexUID",
 				client: customMeili,
-				request: []string{
+				request: []interface{}{
 					"title",
 				},
 			},
 			wantTask: &TaskInfo{
 				TaskUID: 1,
 			},
 		},
+		{
+			name: "TestIndexUpdateFilterableAttributesMixedRawAndObject",
+			args: args{
+				UID:    "indexUID",
+				client: meili,
+				request: []interface{}{
+					"tag",
+					map[string]interface{}{
+						"attributePatterns": []interface{}{"year"},
+						"features": map[string]interface{}{
+							"facetSearch": false,
+							"filter": map[string]interface{}{
+								"equality":   true,
+								"comparison": true,
+							},
+						},
+					},
+				},
+			},
+			wantTask: &TaskInfo{
+				TaskUID: 1,
+			},
+		},
+		{
+			name: "TestIndexUpdateFilterableAttributesOnlyObject",
+			args: args{
+				UID:    "indexUID",
+				client: meili,
+				request: []interface{}{
+					map[string]interface{}{
+						"attributePatterns": []interface{}{"year"},
+						"features": map[string]interface{}{
+							"facetSearch": false,
+							"filter": map[string]interface{}{
+								"equality":   true,
+								"comparison": true,
+							},
+						},
+					},
+				},
+			},
```

Comment:
@ElyarSadig `[]interface` a implicit way, I think after bump to **v1.20** we need use generics for explicit type constraint.